include/rs_utils.h

```diff
@@ -10,24 +10,43 @@ char* rs_replace_all(
 		const char* from,
 		const char* to);
 
+char* rs_consolidate_whitespace(
+		const char* str,
+		const char* whitespace);
+
 void rs_cstring_free(char* str);
 
 class RustString {
-	private:
-		char* str;
-	public:
-		RustString() = delete;
-		RustString( const RustString & ) = delete;
-		RustString( const RustString && ) = delete;
-		explicit RustString( char* String ){
-			str = String;
-		}
-		operator std::string(){
-			return std::string(str);
-		}
-		~RustString() {
-			rs_cstring_free( str );
+private:
+	char* str;
+
+public:
+	RustString() = delete;
+	RustString(const RustString&) = delete;
+
+	RustString(RustString&& rs):
+	str(std::move(rs.str)) {
+		rs.str= NULL;
+	}
+
+	RustString& operator=(RustString&& rs) {
+		if ( &rs != this ) {
+			str = std::move(rs.str);
+			return *this;
 		}
+	}
+
+	explicit RustString(char* ptr) {
+		str = ptr;
+	}
+
+	operator std::string() {
+		return std::string(str);
```

Comment:
Once the object is moved out of, `str` will become `nullptr`, and `std::string` [will have undefined behaviour](https://en.cppreference.com/w/cpp/string/basic_string/basic_string):

> The behavior is undefined if s does not point at an array of at least Traits::length(s)+1 elements of CharT, including the case when s is a null pointer.

Please wrap it in a check.

Reply:
I think I go it.